### PR TITLE
Space-Travellers hub

### PR DESCRIPTION
**Pull request by Divinity-dev**
In this pull request, @tonnytech and @Divinity-dev have implemented the following tasks;
1. We set up the basic dependencies of the project.
2. Created the basic structure for the components
3. We fetched data from the rockets and missions API
4. Displayed this Data on the page.
5. implemented funtions to reserve rockets
6. implemented a funtion to join a mission
7. Displayed The Reserved rockets and missions on a myprofile

### CODE REVIEW

File: src/components/Rocket.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  );
+};
+
+Rocketer.propTypes = {
+  Rocket: PropTypes.shape({
+    rocket_id: PropTypes.string,
+    rocket_name: PropTypes.string,
+    description: PropTypes.string,
+    flickr_images: PropTypes.string,
+    reserved: PropTypes.string,
```
Comment (muneebulrehman) [inline src/components/Rocket.js:46]: - Optional
You are validating the `reserved` prop as a string but a boolean value is being supplied. I would recommend fixing this issue.
image

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good commit history

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -55,234 +23,144 @@ After you're finished please remove all the comments and instructions!
...
-<!-- LIVE DEMO -->
-
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-> Add a link to your deployed project.
-
-- [Live Demo Link](https://google.com)
+-
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
[the remark this answers is not in the document]
Comment (HAMMAS-SALEEM) [reply README.md]: ## :+1:

File: src/__test__/profile.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  //     render(
+  //       <Provider store={store}>
+  //         <MyProfile />
+  //       </Provider>
+  //     );
+
+//     const ProfileContainer = screen.getAllByRole("heading");
+//     // two title headers
+//     expect(ProfileContainer.length).toBe(7);
+//   });
```
[the remark this answers is not in the document]
Comment (HAMMAS-SALEEM) [reply src/__test__/profile.test.js]: ## :+1:

File: src/components/Rocket.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable import/extensions */
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (HAMMAS-SALEEM) [reply src/components/Rocket.js]: ## :+1:

File: src/components/Rockets.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable react/jsx-key */
+/* eslint-disable import/extensions */
```
[the remark this answers is not in the document]
Comment (HAMMAS-SALEEM) [reply src/components/Rockets.js]: ## :+1:

File: src/redux/Rockets/Rocketslice.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable arrow-body-style */
```
[the remark this answers is not in the document]
Comment (HAMMAS-SALEEM) [reply src/redux/Rockets/Rocketslice.js]: ## :+1:

File: src/components/Rockets.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import { useSelector } from 'react-redux';
+import Rocketer from './Rocket';
+
+const Rockets = () => {
+  const { RocketData } = useSelector((state) => state.Rockets);
+  const Rocketlist = RocketData.map((Rock) => (
+    <Rocketer Rocket={Rock} />
```
[the remark this answers is not in the document]
Comment (HAMMAS-SALEEM) [reply src/components/Rockets.js]: ## :+1:

File: README.md [hunk 8]
```diff
@@ -51,11 +51,6 @@
...
```
Comment (HAMMAS-SALEEM) [inline README.md:52]: image

**[OPTIONAL]**

- [ ] Great work on implementing the changes requested by I saw an error in the console window :+1: Kindly fix it :+1: to make your application safe from breaking :+1: You are having props for your Rocket but you didn't pass any props from your parent component :fearful: Kindly fix it by passing the parameter from your parent component :+1:

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @Divinity-dev 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Divinity-dev** [comment]
Thank you 

Sent from Yahoo Mail on Android 
 
  On Sat, Mar 18, 2023 at 7:50 PM, HAMMAS BIN ***@***.***> wrote:   


@HAMMAS-SALEEM approved this pull request.

Status: Approved 🟢 🥇 💯
 


Hey @Divinity-dev 👋
 
Great work on implementing the requested changes. Your project is complete. 😊
There is nothing else to say about your project. 👍
 
Highlights 🥇
 
 ✔️ Descriptive PR Title
 ✔️ Descriptive PR Summary
 ✔️ Following Gitflow
 ✔️ Following Requirements

Optional

All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration.

Cheers and Happy coding! 👏 👏 👏
 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the Code reviews limits policy you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form.

In README.md:
> @@ -51,11 +51,6 @@
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>



[OPTIONAL]
   
   -  Great work on implementing the changes requested by I saw an error in the console window 👍 Kindly fix it 👍 to make your application safe from breaking 👍 You are having props for your Rocket but you didn't pass any props from your parent component 😨 Kindly fix it by passing the parameter from your parent component 👍

**Pull request merged**